src/transformers/modeling_utils.py

```diff
@@ -1043,6 +1043,72 @@ def get_input_embeddings(self) -> nn.Module:
         else:
             raise NotImplementedError
 
+    def warn_if_pad_token_in_input_ids_no_attention_mask(self, input_ids, attention_mask):
+        if attention_mask is not None:
+            # if the attention_mask is defined it's all good
+            return
+
+        if not hasattr(self, "warnings_issued"):
+            self.warnings_issued = {}
+
+        if self.warnings_issued.get("pad_token_in_input_ids", False):
+            # if warning has already been thrown don't throw again
+            return
```

Comment:
This makes the warning appear once, cc @gante . But it's nice if we can re-use the recently added `logger.warning_once`. I will leave @gante to make the final call.